### ISSUE TITLE
React functional component error

### ISSUE THREAD

**Issue by panjiesw** [Swatinem/rollup-plugin-dts#8]
Hi, I'm trying this plugin to bundle a react lib.

I have some function components in different files and would like to bundle it 
with rollup along with its dts. Here is the code of one of the component.

```jsx
export interface AlertProps {
  type: 'info' | 'error';
  children?: React.ReactNode;
}

export const Alert = (props: AlertProps) => {
  const { type, ...rest } = props;
  return (
    <div
      style={{
        background: type === 'info' ? 'blue' : 'red',
      }}
      {...rest}
    />
  );
};
Alert.defaultProps = {
  type: 'info',
};
```

With my `rollup.config.js` file

```js
import { dts } from 'rollup-plugin-dts';

export default [
  otherBundleConfig,
  {
    input: 'src/index.ts',
    output: { file: 'dist/lib.d.ts', format: 'esm' },
    plugins: [dts()],
  },
];
```

It produces error like this:

```
{ kind: 160, code: '\n    (props: AlertProps): JSX.Element;' }
{ kind: 160, code: '\n    (props: ButtonProps): JSX.Element;' }
[!] (dts plugin) Error: Unknown TypeElement
src/Alert.tsx
Error: Unknown TypeElement
    at DeclarationScope.convertMembers (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:261:23)
    at DeclarationScope.convertTypeNode (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:291:25)
    at Transformer.convertVariableStatement (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:485:19)
    at Transformer.convertStatement (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:412:25)
    at new Transformer (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:368:18)
    at Object.transform (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:682:31)
    at Object.transform (<rootProject>/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.js:739:29)
    at <rootProject>/node_modules/rollup/dist/rollup.js:16611:25
```

I tried to change this statement from `rollup-plugin-dts` distribution (I think it comes from `DeclarationScope.ts` file)

```js
// before
if (ts.isMethodDeclaration(node) ||
  ts.isMethodSignature(node) ||
  ts.isConstructorDeclaration(node) ||
  ts.isConstructSignatureDeclaration(node)) {
  this.convertParametersAndType(node);
}

// after
if (ts.isMethodDeclaration(node) ||
    ts.isMethodSignature(node) ||
    ts.isConstructorDeclaration(node) ||
    ts.isCallSignatureDeclaration(node) || // added this condition
    ts.isConstructSignatureDeclaration(node)) {
    this.convertParametersAndType(node);
}
```

Then it works. I'm not experienced with typescript AST though and don't know the implication of that addition.

**Comment by Swatinem** [comment]
Thanks for your report and your solution, which was indeed correct :-)

I released `v0.11.0` with that fix

**Comment by panjiesw** [comment]
Thanks @Swatinem!

While you're at it, please also look at this use case:

```jsx
export function Alert(props: AlertProps) {
  const { type, ...rest } = props;
  return (
    <div
      style={{
        background: type === 'info' ? 'blue' : 'red',
      }}
      {...rest}
    />
  );
}
Alert.defaultProps = {
  type: 'info',
};
```

It should produce this definition:

```ts
export declare function Alert(props: AlertProps): JSX.Element;
export declare namespace Alert {
    var defaultProps: {
        type: string;
    };
}
```

Currently it produces error:

```
{ kind: 244,
  code:
   '\nexport declare namespace Alert {\n    var defaultProps: {\n        type: string;\n    };\n}' }
[!] (dts plugin) Error: unsupported node type
src/Alert.tsx
Error: unsupported node type
    at Transformer.convertStatement (<redacted>/rollup-plugin-dts/dist/rollup-plugin-dts.js:424:19)
    at new Transformer (<redacted>/rollup-plugin-dts/dist/rollup-plugin-dts.js:369:18)
    at Object.transform (<redacted>/rollup-plugin-dts/dist/rollup-plugin-dts.js:683:31)
    at Object.transform (<redacted>/rollup-plugin-dts/dist/rollup-plugin-dts.js:740:29)
    at <redacted>/rollup/dist/rollup.js:16611:25
```

**Comment by Swatinem** [comment]
@panjiesw thanks for the report. I created #9  as separate issue to handle `namespace` declarations, since I think its a little bit more complicated to do that. Will look at it the next few days, not sure yet when I will have time.

**Comment by Swatinem** [comment]
@panjiesw published `v0.12.0` now which should support namespace declarations. Please try that and create a new issue if you still have some showstoppers :-)
